test/e2e-tests.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs the end-to-end tests against eventing-contrib built
+# from source.
+
+# If you already have the *_OVERRIDE environment variables set, call
+# this script with the --run-tests arguments and it will use the cluster
+# and run the tests.
+
+# Calling this script without arguments will create a new cluster in
+# project $PROJECT_ID, start Knative eventing system, install resources
+# in eventing-contrib, run the tests and delete the cluster.
+
+source $(dirname $0)/../vendor/knative.dev/test-infra/scripts/e2e-tests.sh
+
+# If gcloud is not available make it a no-op, not an error.
+which gcloud &> /dev/null || gcloud() { echo "[ignore-gcloud $*]" 1>&2; }
+# Use GNU tools on macOS. Requires the 'grep' and 'gnu-sed' Homebrew formulae.
+if [ "$(uname)" == "Darwin" ]; then
+  sed=gsed
+  grep=ggrep
+fi
+
+# Eventing main config.
+readonly EVENTING_CONFIG="config/"
+
+# NATS Streaming installation config.
+readonly NATSS_INSTALLATION_CONFIG="natss/config/broker/natss.yaml"
+# NATSS channel CRD config directory.
+readonly NATSS_CRD_CONFIG_DIR="natss/config"
+
+# Strimzi installation config template used for starting up Kafka clusters.
+readonly STRIMZI_VERSION="0.11.4"
+readonly STRIMZI_INSTALLATION_CONFIG_TEMPLATE="test/config/100-strimzi-cluster-operator-${STRIMZI_VERSION}.yaml"
+# Strimzi installation config.
+readonly STRIMZI_INSTALLATION_CONFIG="$(mktemp)"
+# Kafka cluster CR config file.
+readonly KAFKA_INSTALLATION_CONFIG="test/config/100-kafka-persistent-single-2.1.0.yaml"
+# Kafka cluster URL for our installation
+readonly KAFKA_CLUSTER_URL="my-cluster-kafka-bootstrap.kafka:9092"
+# Kafka channel CRD config template directory.
+readonly KAFKA_CRD_CONFIG_TEMPLATE_DIR="kafka/channel/config"
+# Kafka channel CRD config template file. It needs to be modified to be the real config file.
+readonly KAFKA_CRD_CONFIG_TEMPLATE="400-kafka-config.yaml"
+# Real Kafka channel CRD config , generated from the template directory and modified template file.
+readonly KAFKA_CRD_CONFIG_DIR="$(mktemp -d)"
+
+function knative_setup() {
+  if is_release_branch; then
+    echo ">> Install Knative Eventing from ${KNATIVE_EVENTING_RELEASE}"
+    kubectl apply -f ${KNATIVE_EVENTING_RELEASE}
+  else
+    echo ">> Install Knative Eventing from HEAD"
+    pushd .
+    cd ${GOPATH} && mkdir -p src/knative.dev && cd src/knative.dev
```

Comment:
could we do a `3rd_party` folder

Reply:
eventing has to be under the `GOPATH`, since we need to use `ko` to install it.